my_Enumerable.rb

```diff
@@ -0,0 +1,153 @@
+  module Enumerable
+    def my_each
+      # your code here
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i])
+      end
+      return self
+    end
+
+    def my_each_with_index
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i],i)
+      end
+      return self
+    end
+
+    def my_select
+      return to_enum(:double) unless block_given?
+      conditionMet = []
+      self.my_each do |item|
+        if yield(item)
+          conditionMet << item
+        end
+      end
+      conditionMet
+    end
+
+    def my_all?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          unless yield(item)
+            return false
+          end
+        else
+          if pattern
+            unless pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_any?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return true
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return true
+            end
+          else
+            if item == nil||item == false
+              return true
+            end
+          end
```

Comment:
### 1. If at least one of the elements of the evaluated collection is not false, it should return true, example:

expected behaviour:

```
irb(main):168:0> arr = [false, false]
=> [false, false]
irb(main):170:0> arr.any?
=> false
irb(main):171:0> arr = [false, true]
=> [false, true]
irb(main):173:0> arr.any?
=> true

```

currently:

```
irb(main):168:0> arr = [false, false]
=> [false, false]
irb(main):169:0> arr.my_any?
=> true
irb(main):171:0> arr = [false, true]
=> [false, true]
irb(main):172:0> arr.my_any?
=> true


```